src/providers/GlobalContext/GlobalContext.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useReducer, createContext } from 'react';
+
+const GlobalContext = createContext();
+
+export function GlobalContextProvider({ children }) {
+  const searchTextReducer = (state, action) => {
```

Comment:
You have two reducers that handle a single action. Ideally you should have one reducer that handles the actions of your application. Merge the two reducers and use a single state. It will keep your application organized, simplified and with a single source of truth.